File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        //Dutch National Flag Algorithm
        int low = 0;
        int mid = 0;
        int high = nums.length - 1;
        
        while(mid <= high) {
            switch(nums[mid]) {
                case(0):
                    int temp = nums[low];
                    nums[low] = nums[mid];
                    nums[mid] = temp;
                    
                    mid++;
                    low++;
                    break;
                case(1):
                    mid++;
                    break;
                case(2):
                    temp = nums[high];
                    nums[high] = nums[mid];
                    nums[mid] = temp;
                    high--;
                    //don't make the mid++
                    break;
                default:
                    break;
            }
        }
    }
}

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        # Dutch National Flag Algorithm
        low = 0
        mid = 0
        high = len(nums) - 1
        
        while(mid <= high):
            if(nums[mid] == 0):
                nums[low],nums[mid] = nums[mid],nums[low]
                low = low + 1
                mid = mid + 1
            elif(nums[mid] == 1):
                mid = mid + 1
            else:
                nums[high],nums[mid] = nums[mid],nums[high]
                high = high - 1
        
        